contracts/plugins/assets/morpho-aave/MorphoTokenisedDeposit.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: BlueOak-1.0.0
+pragma solidity 0.8.19;
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20, IERC20Metadata } from "@openzeppelin/contracts/interfaces/IERC20Metadata.sol";
+import { IMorpho, IMorphoRewardsDistributor, IMorphoUsersLens } from "./IMorpho.sol";
+import { RewardableERC4626Vault } from "../erc20/RewardableERC4626Vault.sol";
+
+struct MorphoTokenisedDepositConfig {
+    IMorpho morphoController;
+    IMorphoRewardsDistributor rewardsDistributor;
+    IERC20Metadata underlyingERC20;
+    IERC20Metadata poolToken;
+    ERC20 rewardToken;
+}
+
+abstract contract MorphoTokenisedDeposit is RewardableERC4626Vault {
```

Comment:
we should write a unit test suite specifically for this contract

Reply:
I will add a suite that tests withdrawals and deposits for all of Morphos markets.